core/src/test/scala/shapeless/typeable.scala

```diff
@@ -499,4 +499,75 @@ class TypeableTests {
     typed[Option[ObjA.type]](c10)
     assertEquals(None, c10)
   }
+
+  trait A
+  trait B
+  class C extends A with B
+
+  @Test
+  def testToString {
+    def typeableString[T](t: T)(implicit tp: Typeable[T]) = tp.toString
+
+    val i: Int = 7
+    assertEquals("Typeable[Integer]", typeableString(i))
+
+    val u: Unit = ()
+    assertEquals("Typeable[BoxedUnit]", typeableString(u))
+
+    val a: Any = ()
+    assertEquals("Typeable[Any]", typeableString(a))
+
+    val av: AnyVal =  7
+    assertEquals("Typeable[AnyVal]", typeableString(av))
+
+    val ar: AnyRef =  ""
+    assertEquals("Typeable[AnyRef]", typeableString(ar))
+
+    val f: Foo = Foo(0, "", true)
+    assertEquals("Typeable[Foo]", typeableString(f))
+
+    val bi: Bar[Int] = Bar(23)
+    assertEquals("Typeable[Bar[Integer]]", typeableString(bi))
+
+    val i1: A with B = new C
+    assertEquals("Typeable[A with B]", typeableString(i1))
+    assertEquals("Typeable[AnyRef with A]", typeableString(new A{}))
```

Comment:
I can drop it by doing something like `def describe = parents.filter(_.describe != "AnyRef") map(_.describe) mkString" with "`. However, it will also filter cases like an explicit `new AnyRef with A{}`. What you think?